tests/pipelines/test_pipelines_text_to_audio.py

```diff
@@ -181,7 +181,9 @@ def run_pipeline_test(self, speech_generator, _):
         outputs = speech_generator("This is a test")
         self.assertEqual(ANY(np.ndarray), outputs["audio"])
 
-        forward_params = {"num_return_sequences": 2, "do_sample": True}
+        forward_params = (
+            {"num_return_sequences": 2, "do_sample": True} if speech_generator.model.can_generate() else {}
+        )
```

Comment:
suggested by @sanchit-gandhi in https://github.com/huggingface/transformers/pull/26285#issuecomment-1729891283